### ISSUE TITLE
Partitions view loads slowly when many runs per partition

### ISSUE THREAD

**Issue by sryza** [dagster-io/dagster#3450]
On a setup with 105 runs per partition, I observed it take 40 seconds to load entries for 30 partitions.  On a similar setup with only a few runs per partition, I observed it take only 4 seconds to load entries for 30 partitions.

While having this many runs per partition may not be the common intended usage, it's very possible that users could get into this state in the same way we did (accidentally submitting a ton of runs).

A potential solution would be to default to only fetching information about the most recent run per partition.  If a user hits the "Show Previous States" checkbox, we could then load information for prior runs.

**Comment by bengotow** [comment]
Hmm this is definitely true - I did some initial performance work on this when we launched this UI with Alex and I think we could go in a couple directions here:

- Make the per-step run stats resolve faster. Currently we're hitting the database with a separate query to resolve the stats of each run, so 40 runs = 40 queries for per-step stats. We should probably invest in something that 'rolls up' these queries and runs a single query for all 40 runs, because this is a problem impacting other parts of Dagit as well.

- Add a graphQL resolver on partition that pre-computes the summary stats for runs on that day. This custom resolver could essentially do the above and get aggregate summary stats more efficiently.

- Fetch only the latest run matching the partition+pipeline filter when "Show Previous States" is unchecked, and remove the per-day run counts shown beneath the boxes. I think this would be pretty good, but the run count is pretty nice.
